modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/util/CommerceAccountHelperImpl.java

```diff
@@ -374,6 +374,28 @@ public AccountEntry getCurrentAccountEntry(
 				return accountEntry;
 			}
 
+			PermissionChecker permissionChecker =
+				PermissionThreadLocal.getPermissionChecker();
+
+			int count =
+				_commerceChannelAccountEntryRelLocalService.
+					getCommerceChannelAccountEntryRelsCount(
+						commerceChannel.getCommerceChannelId(), null,
+						CommerceChannelAccountEntryRelConstants.
+							TYPE_ELIGIBILITY);
+
+			if (permissionChecker.hasPermission(
+					commerceChannelGroupId, AccountEntry.class.getName(),
```

Comment:
why are you checking the channel groupid while in the deleted lines it was passed null?

Reply:
I thought it was a more accurate way to check, since it will check if the user has VIEW_ACCOUNTS permission on the specific group of the channel

Reply:
I'm not sure I've understood the answer, because I don't think we have a way to set specific permissions on groups, but let's see how the non regression tests behave